### ISSUE TITLE
elasticsearch_plugin - unsupported parameter name for module

### ISSUE THREAD

**Issue by ksmets** [ansible/ansible-modules-extras#1785]
##### Ansible Version:

```
ansible 2.0.1.0
  config file = /home/ksmets/my-ansible-playbooks/ansible.cfg
  configured module search path = Default w/o override
```
##### Ansible Configuration:

default
##### Environment:

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.4 LTS"
##### Summary:

Unable to run elasticsearch_plugin
##### Steps To Reproduce:

```
- name: install elasticsearch plugins
  elasticsearch_plugin: name={{ item }} state=present
  with_items:
  - license
  - marvel-agent
```
##### Expected Results:

Install license and marvel-agent plugin
##### Actual Results:

```
ksmets@midgard ~/my-ansible-playbooks> env ANSIBLE_KEEP_REMOTE_FILES=1 ansible-playbook -i production site.yml --tags marvel -K --start-at-task="install elasticsearch plugins"
SUDO password: 

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [bf1]

TASK [marvel : install elasticsearch plugins] **********************************
failed: [bf1] => (item=license) => {"failed": true, "item": "license", "module_stderr": "", "module_stdout": "\r\nTraceback (most recent call last):\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037905.94-19611511873237/elasticsearch_plugin\", line 2127, in <module>\r\n    main()\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037905.94-19611511873237/elasticsearch_plugin\", line 162, in main\r\n    reason = parse_error(out)\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037905.94-19611511873237/elasticsearch_plugin\", line 110, in parse_error\r\n    return string[string.index(reason) + len(reason):].strip()\r\nValueError: substring not found\r\n", "msg": "MODULE FAILURE", "parsed": false}
failed: [bf1] => (item=marvel-agent) => {"failed": true, "item": "marvel-agent", "module_stderr": "", "module_stdout": "\r\nTraceback (most recent call last):\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037907.14-176970966133302/elasticsearch_plugin\", line 2127, in <module>\r\n    main()\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037907.14-176970966133302/elasticsearch_plugin\", line 162, in main\r\n    reason = parse_error(out)\r\n  File \"/home/ksmets/.ansible/tmp/ansible-tmp-1457037907.14-176970966133302/elasticsearch_plugin\", line 110, in parse_error\r\n    return string[string.index(reason) + len(reason):].strip()\r\nValueError: substring not found\r\n", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @site.retry

PLAY RECAP *********************************************************************
bf1                        : ok=1    changed=0    unreachable=0    failed=1  
```

I also tried to run the generated python file on the remote host using both python2 and python3:

```
ksmets@bf1 ~/.a/t/ansible-tmp-1457037256.15-177492007552570> 
python2 elasticsearch_plugin 
Traceback (most recent call last):
  File "elasticsearch_plugin", line 2127, in <module>
    main()
  File "elasticsearch_plugin", line 162, in main
    reason = parse_error(out)
  File "elasticsearch_plugin", line 110, in parse_error
    return string[string.index(reason) + len(reason):].strip()

ksmets@bf1 ~/.a/t/ansible-tmp-1457037256.15-177492007552570> 
python3 elasticsearch_plugin 
{"failed": true, "invocation": {"module_args": {"name": [109, 97, 114, 118, 101, 108, 45, 97, 103, 101, 110, 116], "state": [112, 114, 101, 115, 101, 110, 116]}}, "msg": "unsupported parameter for module: b'name'"}
ValueError: substring not found
```

So it complains about unsupported parameter for module... Any ideas?

**Comment by aaronthebaron** [comment]
I think this is related to issue https://github.com/ansible/ansible-modules-extras/issues/1503

Looks like there is a fix and it's merged in but hasn't shipped out yet.

**Comment by nitzmahone** [comment]
@ksmets Can you verify that this is working in 2.1RC?

**Comment by ksmets** [comment]
It works with v2.1.0.0-0.1.rc1.

**Comment by ksmets** [comment]
Removing plugins doesn't work, yet.

```
TASK [elasticsearch : remove elasticsearch plugins] ****************************
failed: [localhost] (item=marvel-agent) => {"failed": true, "item": "marvel-agent", "msg": "ERROR: Unrecognized option: --timeout\n"}
failed: [localhost] (item=license) => {"failed": true, "item": "license", "msg": "ERROR: Unrecognized option: --timeout\n"}
```

This can be solved by adding the `and state=='present'` to the if timeout conditional:

```
    if timeout and state == 'present':
        cmd_args.append("--timeout %s" % timeout)
```

**Comment by ksmets** [comment]
Also, the `parse_error` function doesn't work as `reason` is not part of the error message returned by elasticsearch-plugin. You might want to change the code to:

```
def parse_error(string):
    reason = "ERROR: "
    return string[string.index(reason) + len(reason):].strip()
```

**Comment by ansibot** [comment]
@ThePixelDeveloper, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by barryib** [comment]
This won't work because when removing plugin the module add unneeded options. ~~This can be solve with #2604. It refactors the module and avoid using unsupported options during plugin installation or removal.~~

Edit: Should be solved with #2839

**Comment by barryib** [comment]
@ThePixelDeveloper ping. I think you're busy actually, but would you like some help to maintain this module?

**Comment by barryib** [comment]
@nitzmahone any updates on this issue?

**Comment by ansibot** [comment]
@ThePixelDeveloper, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ThePixelDeveloper, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ksmets** [comment]
I can confirm that #2839 fixes the issues with (un)installing the plugins
